### ISSUE TITLE
Low priority: The server gets slow when number of users get more than 20

### ISSUE THREAD

**Issue by afshinsadeghi** [OSCOSS/fiduswriter#15]
firstly do a search if its an inherent problem of the python or the framework or its just a library/ server problem, estimate how much time will be needed to correct the issue.

Reported by Alexander:  @aw3s0me

**Comment by aw3s0me** [comment]
Test it with locust.io:
https://github.com/locustio/locust
Tuts: http://savorywatt.com/2014/06/30/load-testing-with-locust-io/

**Comment by afshinsadeghi** [comment]
We had this issue in the last meeting of 17 march 2016.  I and vinay had a discussion and we want that this bug. should be prioritized.  @aw3s0me  check why response time is slow. If every change is stored in the database it have to be replaced with a fast (in memory) queue that reflects of a change in one user to other editors. In parallel the changes can be stored in database periodically.

**Comment by afshinsadeghi** [comment]
But @aw3s0me please do not change the code and socket programming right now. please investigate hoe they could do it in http://etherpad.org/#about and http://blog.etherpad.org/. (no need to scale up to 250 people like them).

**Comment by johanneswilm** [comment]
Hey, too bad I wasn't there. It is already being saved to the database only periodically (every two minutes or if the server is shut down). When it is saved, only the last changes are stores in the database (when it saves). Everything else is in-memory.

We did have a redis-based server part, and I only removed it recently due to too much maintenance work. You could reintroduce redis. But even without it, the server should be able to handle many more than 20 connections. Did you actually have a server slowdown at the meeting, or was there some other issue why person #4 couldn't connect to the server? See my comments about this here [1]. If it does slow down already with 20 users, then you probably need to do something more than just employ redis.

[1] https://github.com/OSCOSS/fiduswriter/issues/18#issuecomment-197679685

**Comment by johanneswilm** [comment]
@sadeghiafshin @aw3s0me @vinaymodi @clange Question: Has it been confirmed that this is about number of users and not about user number X happening to crash the document? If yes, is it happening at 20 users total per server or 20 users per document?

**Comment by johanneswilm** [comment]
Hey, I send out 5300 emails to previous testers, inviting them to test the current version, all within a 4 hour period. I could see lots of new user accounts being created and testing going on, but no slowdown. This was using a micro instance at AWS with a local database (no RDS or any more professional Amazon AWS offerings).

So unless you have other data, it seems to me that there is currently no issue of running a server with 20 users. Can we clsoe this, @sadeghiafshin @aw3s0me @vinaymodi @clange

**Comment by afshinsadeghi** [comment]
Thanks @johanneswilm. Would you please provide the server configuration used for it, so that we apply them in the Gesis and University of Bonn test servers.

**Comment by johanneswilm** [comment]
@sadeghiafshin I updated the Wiki for the repository with an explanation of how it has been set up. It also explains how to add SSL encryption.
